packages/components/src/Button/Button.types.ts

```diff
@@ -121,6 +121,8 @@ interface BasicButtonProps extends ButtonFoundationProps {
    * Used to override the default button role.
    */
   readonly role?: string;
+
+  readonly htmlButtonRef?: React.Ref<HTMLButtonElement>;
```

Comment:
not in love with this name

the reason I've gone with this, and not `ref` via a forwarded ref is because in my mind there _could_ be a distinction to be made between a ref that is associated with the html `button` directly, which is this case

and then another where perhaps we start using an imperativeHandler to add specific Atlantis Button API methods. maybe something like `shake()` who knows.

that said we currently don't offer anything like that

Reply:
I suppose we could still enhance `ref` with more capabilities if we wanted, and then implement a `ref.focus()` which would do the same thing.

that's probably better now that I think about it more. the only thing there is that we've called it `inputRef` on inputs. I wonder why not use `ref` directly for those instances.